File: sdks/go/pkg/models/operations/postgroupsterminatesessions.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type PostGroupsTerminateSessionsRequestBody struct {
	GroupIds []string `json:"group_ids"`
}

type PostGroupsTerminateSessionsRequest struct {
	Retries *utils.RetryConfig
	Request *PostGroupsTerminateSessionsRequestBody `request:"mediaType=application/json"`
}

type PostGroupsTerminateSessionsResponse struct {
	ClientError               *shared.ClientError
	ContentType               string
	SessionTerminationMessage *shared.SessionTerminationMessage
	StatusCode                int64
}

File: sdks/ts/src/sdk/models/operations/postgroupsterminatesessions.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class PostGroupsTerminateSessionsRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=group_ids" })
  groupIds: string[];
}


export class PostGroupsTerminateSessionsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: PostGroupsTerminateSessionsRequestBody;
}


export class PostGroupsTerminateSessionsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  sessionTerminationMessage?: shared.SessionTerminationMessage;

  @SpeakeasyMetadata()
  statusCode: number;
}
